src/vm/dac/CMakeLists.txt

```diff
@@ -1,5 +1,9 @@
-
 include(${CLR_DIR}/dac.cmake)
+include(CMakeDetermineCXXCompiler)
+
+if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND (NOT CMAKE_CXX_COMPILER_VERSION VERSION_LESS 7))
+  set_source_files_properties(../threadpoolrequest.cpp PROPERTIES COMPILE_FLAGS -faligned-new)
```

Comment:
I would prefer setting the flag globally instead of for a single file. Is there a problem with it? Otherwise if we introduced a new place where this flag was needed in the future, we would have to add yet another stuff like this.

Reply:
There seems to be no problem setting it globally. For MSVC, we have the warning disabled in the code file, so I thought to apply the flag for this file only.
Also switched from version sniffing to feature detection for `-faligned-new` flag.